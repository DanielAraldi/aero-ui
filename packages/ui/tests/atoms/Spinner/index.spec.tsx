import { render, screen } from '@testing-library/react-native';

import { Spinner } from '../../../';

describe('<Spinner />', () => {
  describe('Both (iOS and Android)', () => {
    it('Should render Spinner component', () => {
      render(<Spinner useNativeDriver={false} />);
      expect(screen.getByTestId('spinner')).toBeOnTheScreen();
    });

    it("Should render Spinner component with accessibility label as 'Loading' text", () => {
      render(<Spinner useNativeDriver={false} />);
      expect(screen.getByTestId('spinner')).toHaveAccessibleName('Loading');
    });

    it('Should take a snapshot of the Spinner component', () => {
      const component = render(<Spinner useNativeDriver={false} />);
      expect(component).toMatchSnapshot();
    });
  });
});
