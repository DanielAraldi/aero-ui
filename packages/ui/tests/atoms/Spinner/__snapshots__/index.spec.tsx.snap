// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> Both (iOS and Android) Should take a snapshot of the Spinner component 1`] = `
<View
  accessibilityLabel="Loading"
  accessibilityRole="none"
  accessibilityState={
    {
      "disabled": true,
    }
  }
  accessible={true}
  aria-disabled={true}
  aria-label="Loading"
  collapsable={false}
  style={
    {
      "backgroundColor": "transparent",
      "borderBottomColor": "transparent",
      "borderLeftColor": "#000000",
      "borderRadius": 9999,
      "borderRightColor": "transparent",
      "borderTopColor": "transparent",
      "borderWidth": 4,
      "height": 32,
      "transform": [
        {
          "rotate": "0deg",
        },
      ],
      "width": 32,
    }
  }
  testID="spinner"
/>
`;
